dependencies/README.md

```diff
@@ -87,3 +89,80 @@ The tool allows users to choose the scope of entry point classes:
   - **All classes in the input class path**: when reachability check is off, then
     all static linkage errors from all classes in the classpath, regardless of the reachability,
     are reported.
+
+<a name="maven_dependency_graph"></a>
+## Maven Dependency Graph
+
+A Maven dependency graph is a graph data structure where
+- Node: a node is a Maven artifact identified as `groupId:artifactId[:classifier]:version`, where
+  `classifier` is optional.
+- Edge: an (directed) edge is a dependency between Maven artifacts. A dependency from a Maven
+  artifact (_source_ of the dependency) to another artifact (_target_ of the dependency) is defined
+  in the `dependencies` element in pom.xml of a Maven artifact.
+  
+  A dependency has a boolean attribute `optional` and a string attribute `scope`,
+  among other properties listed in [POM Reference: Dependencies][1].
+
+  Self-loops or parallel edges are not possible.
+
+### Graph Construction
+
+Given a ordered list of Maven artifacts, a Maven dependency graph is constructed in the
+following manner:
+
+1. Start with a graph with nodes of the Maven artifacts in the list.
+   The nodes are called _initial nodes_.
+2. Pick up an unvisited node in a graph in breadth-first manner.
+3. By reading dependencies of the node, add new nodes corresponding to the target Maven artifacts,
+   identified by `groupId:artifactId[:classifier]:version`, if not present.
+4. Add edges from the source node to the target nodes of the Maven artifacts.
```

Comment:
This doesn't sound quite right. You're not constructing the edges until all nodes are found, but then how do you find the nodes?

Reply:
Updated step 3 to clarify the 2 and 3 are tied together.